### ISSUE TITLE
The *.tar.gz releases are missing 'version' in composer.json

### ISSUE THREAD

**Issue by pradtke** [simplesamlphp/simplesamlphp#794]
The SSP releases packaged as `*.tar.gz` do not include `version` in the `composer.json`.
This prevents `composer` from knowing what version of SSP is installed when satisfying dependencies.
For example I have a library that has been updated to use SSP 1.15 so it has a 
```json
"require": {
"simplesamlphp/simplesamlphp": "~1.15"
}
```

however if I go to do a `composer require mymodule` in an SSP 1.15 installation I get an error

```
 Problem 1
    - Installation request for simplesamlphp/simplesamlphp No version set (parsed as 1.0.0) -> satisfiable by simplesamlphp/simplesamlphp[No version set (parsed as 1.0.0)].
    - cirrusidentity/simplesamlphp-internal-utils dev-master requires simplesamlphp/simplesamlphp >=1.15 -> satisfiable by simplesamlphp/simplesamlphp[v1.15.0, v1.15.1, v1.15.2].
    - Can only install one of: simplesamlphp/simplesamlphp[v1.15.0, No version set (parsed as 1.0.0)].
    - Can only install one of: simplesamlphp/simplesamlphp[v1.15.1, No version set (parsed as 1.0.0)].
    - Can only install one of: simplesamlphp/simplesamlphp[v1.15.2, No version set (parsed as 1.0.0)].
```

Adding "version" into composer.json fixes this issue.

**Comment by jaimeperez** [comment]
Hi Patrick!

Why is this an issue now? Is it a new requirement? It has worked perfectly fine up until now...

**Comment by pradtke** [comment]
This is the first time I've (and AFAIK anyone) has restricted which version of SSP a module can be installed in. SSP 1.15 has enough features (like namespace support for classes in modules)/changes that my module will not work in version 1.14 of SSP - so in my module's composer.json file I say it requires version 1.15 of SSP.

If I put no restrictions on the SSP version the module can be installed into then you are correct - it installs fine - however people would be able to install into 1.14 where it wouldn't work correctly.

**Comment by jaimeperez** [comment]
Uhm, that's interesting, because we're requiring a specific version of SimpleSAMLphp in one of our modules. However, we specify a single commit, not a version, so that might be the issue.

In any case, duly noted 😉

**Comment by jaimeperez** [comment]
Fixed in 1.15.3.

**Comment by pradtke** [comment]
Thanks @jaimeperez , though I wonder if `build-release.sh` is a better place for this and to do a
`php "$TARGET/composer.phar" config version $VERSION ` and then the change would be isolated to the *.tar.gz.  I'm not familiar enough with composer internals to know what affect putting the version into the vcs composer.json will have since composer says its smart enough to determine version from tags/commits/branches- e.g. does it prevent someone from requiring `dev-master` or a specific commit
